### PR TITLE
ESLint (and ESLint-related dependencies) upgrade

### PR DESCRIPTION
**Pull request by codeheroics**
* Upgrade of every ESLint-related dependencies
* Modifications based on the rules modified by the upgrade
* I took the opportunity to change a few things that had been bothering me for a while (I hate the default `react/sort-comp ` rule we inherit from the airbnb preset)
* Disabled some other rules (mainly `react/forbid-prop-types`), we should later look into activating it

Totally open to modifying some more rules, I just wanted to do the upgrades / get rid of the warnings

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
-    }
+    'react/sort-comp': [2, {
+      order: [
+        'static-methods',
+        'lifecycle',
+        'everything-else',
+        'render'
+      ]
+    }],
+    'jsx-a11y/accessible-emoji': 0,
```
Comment (Fenntasy) [inline .eslintrc.js]: Uh? We can't use emojis without aria roles, in mobile?

File: src/components/FilterList.js [hunk 2]
```diff
@@ -31,14 +31,13 @@ const styles = StyleSheet.create({
...
-export class FilterList extends Component {
+export default class FilterList extends Component {
   static propTypes = {
     contents: PropTypes.array,
     onSelect: PropTypes.func,
-    title: PropTypes.string,
   }
 
-  static defaultProps = { contents: [], title: '' }
+  static defaultProps = { contents: [], onSelect: () => {} }
```
Comment (Fenntasy) [inline src/components/FilterList.js]: best function ever.

**Comment by Fenntasy** [comment]
Well... I can live with or without parenthesis for arrow functions with one argument. But I can only live with _one_ choice for them ^^'
You added some and removed others, I can't even.

File: .eslintrc.js (hunk 1 above)
Comment (codeheroics) [inline .eslintrc.js]: Yep, it's a jsx-a11y check activated by the config we inherit from eslint-config-airbnb, except I guess the presets are mostly for web React ^^'

**Comment by codeheroics** [comment]
Parenthesesgate : It's the default eslint-config-airbnb preset =D something like "if there are multiple lines to the function body, add parentheses" :) I'm going to change it for a simpler rule :)

**Comment by codeheroics** [comment]
Parenthesesgate : It's the default eslint-config-airbnb preset =D something like "if there are multiple lines to the function body, add parentheses" :) I'm going to change it for a simpler rule :)

**Comment by codeheroics** [comment]
Parenthesesgate : Rule changed to
`    'arrow-parens': [2, 'as-needed']`
when it had previously a `"requireForBlockBody": true` option.

I then `eslint --fix`ed everything :)

**Comment by Fenntasy** [comment]
I want to merge but there are conflicts :(

**Comment by codeheroics** [comment]
rebased!

**Pull request merged**